### PR TITLE
Cot cleanup

### PR DESCRIPTION
**Pull request by escapewindow**
While writing up the CoT technical details doc, we noticed some things we could clean up.

- remove the unused `extra_run_task_arguments` config
- clarify some path logic
- remove the obsolete `check_num_tasks` -- because of the logic in adding parent dirs per task, we're guaranteed at least one decision task in the graph

@tomprince do you feel up to reviewing this?

I filed #397 for dropping non-hook action support. I was going to bundle that here, but the test changes were large enough that it deserves its own PR.

### CODE REVIEW

File: scriptworker/artifacts.py [hunk 1]
```diff
@@ -424,3 +429,22 @@ def get_optional_artifacts_per_task_id(upstream_artifacts):
             )
 
     return optional_artifacts_per_task_id
+
+
+def assert_is_parent(path, parent_dir, exception=ScriptWorkerTaskException):
```
Comment (JohanLorenzo) [inline scriptworker/artifacts.py]: What's the use case for passing down `exception`? I don't see it used anywhere in this PR. I wonder if it's case of YAGNI.

File: scriptworker/test/test_artifacts.py [hunk 2]
```diff
@@ -377,3 +378,31 @@ def test_get_single_upstream_artifact_full_path(context):
...
+@pytest.mark.parametrize("path, parent_path, raises", ((
+    "/foo/bar/baz", "/foo/bar", False
+), (
+    "/foo", "/foo/bar", True
+), (
+    "/foo/bar/..", "/foo/bar", True
+)))
+def test_assert_is_parent(path, parent_path, raises):
+    if raises:
+        with pytest.raises(ScriptWorkerTaskException):
```
Comment (JohanLorenzo) [inline scriptworker/test/test_artifacts.py]: Nit: assert on `exception` if we want to support this use case.

File: scriptworker/context.py [hunk 3]
```diff
@@ -82,13 +83,22 @@ def claim_task(self, claim_task):
...
             self.temp_credentials = None
             self.task = None
 
+    def verify_task(self):
+        """Run some task sanity checks on ``self.task``."""
+        for upstream_artifact in self.task.get("payload", {}).get("upstreamArtifacts", []):
+            task_id = upstream_artifact["taskId"]
+            for path in upstream_artifact["paths"]:
+                if os.path.isabs(path) or '..' in path:
+                    raise CoTError("upstreamArtifacts taskId {} has illegal path {}!".format(task_id, path))
```
Comment (JohanLorenzo) [inline scriptworker/context.py]: 👍

File: scriptworker/test/test_artifacts.py [hunk 4]
```diff
@@ -377,3 +378,31 @@ def test_get_single_upstream_artifact_full_path(context):
...
+            swartifacts.assert_is_parent(path, parent_path)
+    else:
+        swartifacts.assert_is_parent(path, parent_path)
+
+
+def test_assert_is_parent_softlink(tmpdir):
```
Comment (JohanLorenzo) [inline scriptworker/test/test_artifacts.py]: 👍  for thinking of soft links. That makes me think, should we also test hard links? https://docs.python.org/2/library/os.html#os.link

**Comment by escapewindow** [comment]
> Looks great to me! Thanks for these cleanups.
Thanks!

> Do you think we should use `assert_is_parents` at the these places too?
> 
> https://github.com/mozilla-releng/scriptworker/blob/0d2053f600b5b3224db916a08c1349eb4db6f863/scriptworker/cot/verify.py#L2168

This is only for the create-test-workdir tool, which isn't used in production.

> https://github.com/mozilla-releng/scriptworker/blob/0d2053f600b5b3224db916a08c1349eb4db6f863/scriptworker/version.py#L75

This is only used to find the `version.json` file.

I don't think either of these are concerning, but I might be missing something.

File: scriptworker/test/test_artifacts.py (hunk 4 above)
Comment (escapewindow) [reply scriptworker/test/test_artifacts.py]: I don't think we can download a hardlink, while we might be able to download a softlink.

File: scriptworker/artifacts.py (hunk 1 above)
Comment (escapewindow) [reply scriptworker/artifacts.py]: Hm. I think I was angling for raising a `CoTError` if we hit this during `cot.verify`, and a `ScriptWorkerTaskException` elsewhere, but it looks like I don't use it.

**Pull request merged**